### ISSUE TITLE
Produce binaries of `prqlc`

### ISSUE THREAD

**Issue by max-sixty** [PRQL/prql#1657]
Currently installing `prqlc` requires a rust compiler. This also means it takes some time to install it with brew.

Some projects compile and publish binaries. Then `brew install` is really fast. One example is [tealdeer](https://github.com/dbrgn/tealdeer/releases/tag/v1.6.1). Here's their [brew formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tealdeer.rb), and their [release script](https://github.com/dbrgn/tealdeer/blob/main/.github/workflows/release.yml). 

I think this approach is called a "bottle" in brew lang.

If we could implement this for PRQL, that would be great, and make it more accessible for folks to download and install the compiler, as well as signaling that we care about that easy-ness.

**Comment by snth** [comment]
`prql-query` compiles binary artifacts as part of the releases, see [prql-query/releases](https://github.com/PRQL/prql-query/releases). 

It's all handled by the [taiki-e/upload-rust-binary-action](https://github.com/taiki-e/upload-rust-binary-action) Github Action. See [prql-query release GHA](https://github.com/PRQL/prql-query/blob/main/.github/workflows/release.yaml).

[tealdear](https://github.com/dbrgn/tealdeer/releases/tag/v1.6.1)'s matrix of target platforms is more impressive so it would be great if we could adopt their approach.

What I haven't been able to figure out is how to do the brew "bottleing". I thought that had to happen on the brew core side but it would be great if we could do that ourselves. @pthariensflame is that something you are knowledgeable in and could assist us with since you so kindle helped us in the [homebrew-prql](https://github.com/PRQL/homebrew-prql) repo?

Alternatively, and for people who don't have `brew`, I provide this rather unwieldy command in the prql-query README:
```sh
VERSION=v0.0.14 wget https://github.com/prql/prql-query/releases/download/$VERSION/pq-x86_64-unknown-linux-gnu.tar.gz && \
    tar xvzf pq-x86_64-unknown-linux-gnu.tar.gz --directory ~/.local/bin && \
    rm pq-x86_64-unknown-linux-gnu.tar.gz
```
Could we maybe wrap that in a script and host it on the https://prql-lang.org/ website, e.g. at https://prql-lang.org/get-prqlc and https://prql-lang.org/get-pq, like https://get.docker.com// ?

**Comment by pthariensflame** [comment]
Homebrew will bottle things themselves if they're in one of the official taps, but for a custom tap like this you have to handle it yourself, yeah. Unfortunately I'm not knowledgeable enough to be able to do that.  ☹️

**Comment by max-sixty** [comment]
> Homebrew will bottle things themselves if they're in one of the official taps

Ah great! So maybe we should prioritize https://github.com/PRQL/prql/issues/866 and deprecate our own tap, at least for `prqlc`...

**Comment by max-sixty** [comment]
> Could we maybe wrap that in a script and host it on the https://prql-lang.org/ website, e.g. at https://prql-lang.org/get-prqlc and https://prql-lang.org/get-pq, like https://get.docker.com// ?

For sure, could `curl` a script from there

**Comment by snth** [comment]
This should be useful (for the non-homebrew part): https://news.ycombinator.com/item?id=34614497

**Comment by snth** [comment]
> Ah great! So maybe we should prioritize https://github.com/PRQL/prql/issues/866 and deprecate our own tap, at least for prqlc...

IIRC when when I set up the prql-query tap, prql-compiler was already part of homebrew core and `brew install` already downloaded the binary for my platform. That might have changed with the switch to `prqlc`. I get very confused with the whole brew ecosystem so can't be much help there unfortunately.

**Comment by max-sixty** [comment]
> This should be useful (for the non-homebrew part): https://news.ycombinator.com/item?id=34614497

This could be great!

**Comment by vanillajonathan** [comment]
Binaries can also be included in releases on GitHub.
https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository?tool=webui#creating-a-release

**Comment by eitsupi** [comment]
Closed by #2578
